tidy3d/components/file_util.py

```diff
@@ -28,3 +28,40 @@ def extract_gzip_file(input_gz_file, output_file):
     with gzip.open(input_gz_file, "rb") as file_in:
         with open(output_file, "wb") as file_out:
             shutil.copyfileobj(file_in, file_out)
+
+
+def replace_values(
+    values: list | dict | tuple, search_value: str | int | object, replace_value: str | int | object
+) -> list | dict | tuple:
+    """
+    Recursively replace all None values with an empty list in a nested dictionary.
+    Can enter tuples and continue iterating through dicts stored inside.
+    Could be modified to be a generic search and replace for nested dicts.
+
+    Parameters
+    ----------
+    values :
+        The input list/dict/tuple object to iterate through.
+    searchValue :
+        A string/int/object to look for in values.
+    replaceValue :
+        A string/int/object to replace the value in values.
+
+    Returns
+    -------
+    list/dict/tuple
+        values type object with searchValue terms replaced by replaceValue.
+    """
+    try:
+        if values == search_value:
+            return replace_value
+    except ValueError:  # Used to catch arrays which cannot be evaluated with ==
```

Comment:
python is complaining because it wants `np.all()` or `np.any()` here? I think it wouldn't be unreasonable to just throw a `np.all(values == search_value)` in here. and not try/except. We probably dont want this failing silently.